src/main/java/com/joyent/manta/cosbench/MantaStorage.java

```diff
@@ -209,15 +224,22 @@ public void createObject(final String container,
         } else {
             contentLength = length;
         }
-
         MantaHttpHeaders headers = new MantaHttpHeaders();
 
         try {
             if (durabilityLevel != null) {
                 headers.setDurabilityLevel(durabilityLevel);
             }
-
-            client.put(path, data, contentLength, headers, null);
+            logger.info("Multipart : /{} {} ", multipart, client.getContext().isClientEncryptionEnabled());
+            if (this.multipart) {
+                if (client.getContext().isClientEncryptionEnabled()) {
+                    multipartUpload(data, path, new EncryptedServerSideMultipartManager(client));
+                } else {
+                    multipartUpload(data, path, new ServerSideMultipartManager(client));
```

Comment:
I'd recommend constructing a `MantaMultipartManager` (encrypted or not) during `init`. Users are expected to share a `MultipartManager` the same way they share a `MantaClient` (plus there's some reflection in the constructors that would mess with cosbench timings)